File: Data Structures & Algorithms/duplicate-integer/submission-5.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        /**
        Time: O(nlogn)
        Space: O(1)
        */

        Arrays.sort(nums);

        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] == nums[i + 1]) {
                return true;
            }
        }
        return false; 
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        """
        Given: 
        - nums.length >= 0
        - not uique numbers
        - neg or pos integers
        - Time: O(n)

        Return: longest consecutive sequences's length

        """

        if len(nums) <= 1:
            return len(nums)
        
        nums.sort()

        res = 0
        count = 0

        for i in range(len(nums) - 1):
            if nums[i + 1] == nums[i]:
                continue
            elif nums[i + 1] - nums[i] == 1:
                count += 1
                res = max(res, count)
            else:
                res = max(res, count)
                count = 0

        return res + 1
            